common/types.go

```diff
@@ -116,10 +131,22 @@ func (h Hash) Generate(rand *rand.Rand, size int) reflect.Value {
 	return reflect.ValueOf(h)
 }
 
+// EmptyHash returns if a hash is empty or not
 func EmptyHash(h Hash) bool {
 	return h == Hash{}
 }
 
+// Scan allows scanning from (byte slice) to *Hash
+func (h *Hash) Scan(src interface{}) error {
```

Comment:
Hash and Address are arrays. If these types are `Scanner` I expect them also to be a `Valuer`?

Reply:
Surprisingly, I didn't have to set a Value, let me check why!

Reply:
Ok, I know why: I always use Address.Bytes() when I bind the parameters.

Reply:
This PR might not be perfect, but it is at least providing a way out. Right now, I have to encapsulate the common.Address and Hash into another struct, to define the Scan func.

Reply:
Let me know if you want me to go full on, with Valuer as well as a Scan taking into account that not everyone will wanna save the data in a Binary form (even though, that's probably the best)

Reply:
Could you add the Valuer method so that users can use both types directly without the need for `Bytes()`? Storing it by default as a byte slice and let the driver work out the details is imho fine. If someone wants to store it in a different format they can wrap it.

It would be nice to separate the documentation and code changes in separate commits.